src/index.scss

```diff
@@ -1,9 +1,10 @@
+@import 'carbon-components/scss/globals/scss/styles.scss';
+
 $feature-flags: (
```

Comment:
This should come before the styles import. It is affecting your layout.

See https://www.carbondesignsystem.com/tutorial/react/step-2#install-grid